app/models/view/alerts/Alert.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package models.view.alerts;
+
+import com.arpnetworking.logback.annotations.Loggable;
+import com.google.common.collect.ImmutableMap;
+import models.internal.alerts.AlertEvaluationResult;
+import models.internal.scheduling.JobExecution;
+
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * View model for an alert.
+ * <p>
+ * Play view models are mutable.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+@Loggable
+public final class Alert {
+    private UUID _id;
+    private String _name;
+    private String _description;
+    private boolean _enabled;
+    private ImmutableMap<String, Object> _additionalMetadata;
+    private AlertFiringState _firingState;
+
+    /**
+     * Construct a view model from its internal representation.
+     *
+     * @param internal The internal model for this alert.
+     * @param mostRecentEvaluation The most recent evaluation result for this alert.
+     * @return An alert view model.
+     */
+    public static Alert fromInternal(
```

Comment:
Historically, these methods were not on the view model classes to avoid the server-side rendering code (view code) from having dependencies on types in the internal model.

Reply:
(it's been a while since I've been in this codebase, so I'm not certain whether this paradigm changed with reporting/alerting)

Reply:
It may have changed, I did it this way following what was done with reports:
https://github.com/ArpNetworking/metrics-portal/blob/master/app/models/view/reports/Report.java#L118

Reply:
I'm starting to see a pattern here.